src/app/app.module.ts

```diff
@@ -36,6 +36,7 @@ import { PageNotFoundComponent } from './page-not-found';
 import { FlashMessagesModule } from 'angular2-flash-messages';
 import { UsersComponent,
          UserDetailComponent } from './users';
+import { GeneralinfoComponent } from './dataset/generalinfo/generalinfo.component';
```

Comment:
Suggestion (more a question of taste).

Place the following line in `app/dataset/index.ts`
```
export * from './datasetform/generalinfo/generalinfo..component';
```

and use instead:
```
import { DatasetComponent,
    DatasetformComponent,
    DatasetDirective,
    GeneralInfoComponent } from './dataset';
```
That way you can keep all of the dataset-related stuff together.